e_os.h

```diff
@@ -94,149 +88,13 @@ extern "C" {
-        * some module, you'd need to override _WIN32_WINNT definition in
-        * the target module in order to "reach for" prototypes, but replace
-        * calls to new functions with indirect calls. Alternatively it
-        * might be possible to achieve the goal by /DELAYLOAD-ing .DLLs
-        * and check for current OS version instead.
-        */
-#    define _WIN32_WINNT 0x0501
-#   endif
-#   if defined(_WIN32_WINNT) || defined(_WIN32_WCE)
-       /*
-        * Just like defining _WIN32_WINNT including winsock2.h implies
-        * certain "discipline" for maintaining [broad] binary compatibility.
-        * As long as structures are invariant among Winsock versions,
-        * it's sufficient to check for specific Winsock2 API availability
-        * at run-time [DSO_global_lookup is recommended]...
-        */
-#    include <winsock2.h>
-#    include <ws2tcpip.h>
-       /* yes, they have to be #included prior to <windows.h> */
-#   endif
-#   include <windows.h>
-#   include <stdio.h>
-#   include <stddef.h>
-#   include <errno.h>
-#   if defined(_WIN32_WCE) && !defined(EACCES)
-#    define EACCES   13
-#   endif
-#   include <string.h>
-#   ifdef _WIN64
-#    define strlen(s) _strlen31(s)
-/* cut strings to 2GB */
-static __inline unsigned int _strlen31(const char *str)
-{
-    unsigned int len = 0;
-    while (*str && len < 0x80000000U)
-        str++, len++;
-    return len & 0x7FFFFFFF;
-}
-#   endif
-#   include <malloc.h>
-#   if defined(_MSC_VER) && !defined(_WIN32_WCE) && !defined(_DLL) && defined(stdin)
-#    if _MSC_VER>=1300 && _MSC_VER<1600
-#     undef stdin
-#     undef stdout
-#     undef stderr
-FILE *__iob_func();
-#     define stdin  (&__iob_func()[0])
-#     define stdout (&__iob_func()[1])
-#     define stderr (&__iob_func()[2])
-#    elif _MSC_VER<1300 && defined(I_CAN_LIVE_WITH_LNK4049)
-#     undef stdin
-#     undef stdout
-#     undef stderr
-         /*
-          * pre-1300 has __p__iob(), but it's available only in msvcrt.lib,
-          * or in other words with /MD. Declaring implicit import, i.e. with
-          * _imp_ prefix, works correctly with all compiler options, but
-          * without /MD results in LINK warning LNK4049: 'locally defined
-          * symbol "__iob" imported'.
-          */
-extern FILE *_imp___iob;
-#     define stdin  (&_imp___iob[0])
-#     define stdout (&_imp___iob[1])
-#     define stderr (&_imp___iob[2])
-#    endif
-#   endif
-#  endif
-#  include <io.h>
-#  include <fcntl.h>
-
-#  ifdef OPENSSL_SYS_WINCE
-#   define OPENSSL_NO_POSIX_IO
-#  endif
-
-#  define EXIT(n) exit(n)
-#  define LIST_SEPARATOR_CHAR ';'
-#  ifndef W_OK
-#   define W_OK        2
-#  endif
-#  ifndef R_OK
-#   define R_OK        4
-#  endif
-#  ifdef OPENSSL_SYS_WINCE
-#   define DEFAULT_HOME  ""
-#  else
-#   define DEFAULT_HOME  "C:"
-#  endif
+# if defined(WIN32) || defined(WINDOWS) || defined(MSDOS)
+#  include "include/internal/plat-msft.h"
```

Comment:
You do know that we have a os-dep directory for this kind of thing, right?

Reply:
Yes, but this seems different.  Should e_os include os-dep files?  Looking at the one file that's there, it seems "special" purpose.  But I am willing to change if you think so.

Reply:
The haiku.h file is included through compiler flags, it's true, but I think it's kind of a special case (unless such practice is common among compilers these days).

We have talked long ago about having an OS dependent set of source files, we just haven't really started that work yet, except for the haiku header file.  So I'd say that using that same space for more OS dependent stuff, and I see no real reason why `e_os.h` couldn't be used to include them.  Ultimately, `e_os.h` could become a bunch of conditional includes of files in `os-dep`...

So yes, I think so.